src/s390x/zipl.rs

```diff
@@ -95,9 +86,51 @@ pub fn install_bootloader<P: AsRef<Path>>(
 
     runcmd!("zipl", "--blsdir", blsdir, "--config", conffile.path())?;
 
-    if let Some(disk) = disk {
-        eprintln!("Updating re-IPL device");
-        runcmd!("chreipl", disk)?;
-    }
     Ok(())
 }
+
+/// Returns the first-boot kargs embedded in the contents `s` of a firstboot file.
+///
+/// Note this isn't intended to be a general purpose GRUB config language parser. Only the exact
+/// format used by coreos-installer is recognized. Any other format triggers an error.
+fn extract_firstboot_kargs(s: &str) -> Result<Option<String>> {
+    let s = s.trim();
+    if s.is_empty() {
+        return Ok(None);
+    }
+
+    let captures = Regex::new(r#"^set ignition_network_kcmdline="([^\n]*)"$"#)
+        .expect("compiling RE")
+        .captures(&s)
+        .context("couldn't parse kargs from ignition.firstboot file")?;
+    match captures.get(1).expect("kargs").as_str() {
+        "" => Ok(None), // this shouldn't really happen, but be nice
+        kargs => Ok(Some(kargs.into())),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_extract_firstboot_kargs() {
+        assert_eq!(extract_firstboot_kargs("").unwrap(), None);
+        assert_eq!(extract_firstboot_kargs("\n").unwrap(), None);
+        assert_eq!(
+            extract_firstboot_kargs("set ignition_network_kcmdline=\"\"").unwrap(),
+            None
+        );
+        assert_eq!(
+            extract_firstboot_kargs("set ignition_network_kcmdline=\"foobar\"").unwrap(),
+            Some("foobar".into())
+        );
+        assert_eq!(
+            extract_firstboot_kargs("set ignition_network_kcmdline=\"foo bar\"").unwrap(),
+            Some("foo bar".into())
+        );
+        assert!(extract_firstboot_kargs("set ignition_network_kcmdline=\"\n\"").is_err());
+        assert!(extract_firstboot_kargs("set ignition_network_kcmdline=\"foo\nbar\"").is_err());
+        assert!(extract_firstboot_kargs("foobar").is_err());
```

Comment:
Other things to possibly test:
- Valid `set` line followed by a newline
- Valid `set` line preceded by something else
- Valid `set` line followed by something else